admin/model/extension/payment/razorpay.php

```diff
@@ -0,0 +1,281 @@
+        }
+
+        $sort_data = array(
+        's.subscription_id',
+        's.created_at',
+        's.status'
+        );
+
+        if (isset($data['sort']) && in_array($data['sort'], $sort_data)) {
+            $sql .= " ORDER BY " . $data['sort'];
+        } else {
+            $sql .= " ORDER BY p.entity_id";
+        }
+
+        if (isset($data['order']) && ($data['order'] == 'DESC')) {
+            $sql .= " DESC";
+        } else {
+            $sql .= " ASC";
+        }
+
+        $query = $this->db->query($sql);
+        return $query->rows;
+    }
+    public function getTotalSubscriptions($data = array())
+    {
+        $sql = "SELECT COUNT(*) AS total FROM `" . DB_PREFIX . "razorpay_subscriptions` s";
+        
+        $sql .=" LEFT JOIN " . DB_PREFIX . "razorpay_plans p ON (p.entity_id = s.plan_entity_id)";
+        $sql .=" LEFT JOIN " . DB_PREFIX . "product_description op ON (op.product_id = p.opencart_product_id)";
+        $sql .=" LEFT JOIN " . DB_PREFIX . "customer c ON (s.opencart_user_id = c.customer_id )";
+
+        if (!empty($data['filter_subscription_id'])) {
+            $implode = array();
+
+            $sub_statuses = explode(',', $data['filter_subscription_id']);
+
+            foreach ($sub_statuses as $sub_id) {
+                $implode[] = "subscription_id = '" . $sub_id . "'";
+            }
+
+            if ($implode) {
+                $sql .= " WHERE (" . implode(" OR ", $implode) . ")";
+            }
+        } elseif (isset($data['filter_plan_name']) && $data['filter_plan_name'] !== '') {
+            $sql .= " WHERE plan_id = '" . $data['filter_plan_name'] . "'";
+        } else {
+            $sql .= " WHERE plan_id > '0'";
+        }
+        if (!empty($data['filter_date_created'])) {
+            $sql .= " AND DATE(s.created_at) = DATE('" . $this->db->escape($data['filter_date_created']) . "')";
+        }
+
+        $query = $this->db->query($sql);
+        return $query->row['total'];
+    }
+    public function getSubscriptionInfo($entity_id)
+    {
+        $sql = "SELECT s.*,s.entity_id as sub_id,s.status as sub_status,s.created_at as sub_created,p.*,op.name,c.firstname,c.lastname FROM `" . DB_PREFIX . "razorpay_subscriptions` s";
+        $sql .=" LEFT JOIN " . DB_PREFIX . "razorpay_plans p ON (p.entity_id = s.plan_entity_id)";
+        $sql .=" LEFT JOIN " . DB_PREFIX . "product_description op ON (op.product_id = p.opencart_product_id)";
+        $sql .=" LEFT JOIN " . DB_PREFIX . "customer c ON (s.opencart_user_id = c.customer_id )";
+        
+        $sql .= " WHERE s.entity_id= '" .$entity_id . "'";
+        $query = $this->db->query($sql);
+        return $query->row;
+    }
+    public function resumeSubscription($entity_id,$updated_by)
+    {
+        $this->db->query("UPDATE " . DB_PREFIX . "razorpay_subscriptions SET status = 'active',updated_by = '".$updated_by. "' WHERE entity_id = '" .$entity_id . "'");
+    }
+    public function pauseSubscription($entity_id)
+    {
```

Comment:
$updated_by is not passed here but used in query for `pauseSubscription`, `cancelSubscription` functions